### PR TITLE
Add supports for hapi 17

### PR DESCRIPTION
**Pull request by joseluisdiaz**
Support HAPI 17 through a factory method.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +82,17 @@ agent.errorReporter.captureException('My error', {
 });
 ```
 
+For `hapi` version 17 and above there's a specific plugin for this. You may satup this using:
```
Comment (dafortune) [inline README.md]: /s/satup/setup/

File: README.md [hunk 2]
```diff
@@ -82,6 +82,17 @@ agent.errorReporter.captureException('My error', {
...
+
+```js
+var hapi = require('hapi');
+var server = new hapi.Server();
+agent.init(pkg, env);
+
+await server.register(agent.errorReporter.hapi.pluginV17);
```
Comment (dafortune) [inline README.md]: Isn't this a bit weird? I understand the idea of supporting both, and is a detail, but looks kind of weird to me.

File: lib/error_reporter.js [hunk 3]
```diff
@@ -6,30 +6,57 @@ module.exports = function(pkg, env) {
...
+
+      // extract err, id depends on the hapi version.
+      // for >= hapi17:
+      //            - event has a property error
+      //            - the id is on `request.info.id`
+      // for < hapi17:
+      //            - event is the directly the error
+      //            - the id is on `request.id`
+      const err = event.isBoom ? event : event.error;
+      const id = request.id ? request.id : request.info.id;
```
Comment (dafortune) [inline lib/error_reporter.js]: This is prone to an uncaught (cannot read property id of undefined) if for some reason the property info is not present, are you sure it is always there for hapi v17?

File: test/stubs.test.js [hunk 4]
```diff
@@ -31,6 +31,13 @@ describe('stubs', function() {
       done();
     });
 
+    it('should have a hapi v17 plugin', function(done) {
```
Comment (dafortune) [inline test/stubs.test.js]: Is it the only kind of test we need? I mean, what about the new way to capture errors, etc?

File: README.md (hunk 2 above)
Comment (dafortune) [reply README.md]: OK, got it there is a hapi.plugin.

File: lib/error_reporter.js (hunk 3 above)
Comment (joseluisdiaz) [reply lib/error_reporter.js]: That's from the spec!

File: test/stubs.test.js (hunk 4 above)
Comment (joseluisdiaz) [reply test/stubs.test.js]: I wanted to only replicate the same test that we have right now; we could add a few test in a different PR.

**Pull request merged**